modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1620,11 +1660,29 @@ private BaseModel<ExternalReferenceCodeModel> _getManyToOneRelatedModel(
 			objectRelationship.getObjectRelationshipId(), primaryKey);
 	}
 
-	private String _getNewValue(String value) {
-		return StringBundler.concat(
-			value, StringPool.SPACE, StringPool.OPEN_PARENTHESIS,
-			_language.get(LocaleUtil.getSiteDefault(), "copy"),
+	private String _getNewValue(
+		String fieldName, ObjectDefinition objectDefinition, String value) {
+
+		String copy = _language.get(LocaleUtil.getSiteDefault(), "copy");
+
+		String newValue = StringBundler.concat(
+			value, StringPool.SPACE, StringPool.OPEN_PARENTHESIS, copy,
 			StringPool.CLOSE_PARENTHESIS);
+
+		String prefix = value;
+
+		for (int i = 1;; i++) {
+			SearchResponse searchResponse = _getSearchResponse(
```

Comment:
Should this feature only work with localized Object Definition fields? We support object-level and field-level localization settings.
I imagine you used Elasticsearch for performance reasons, however, for it to work for non-indexed objects, you would have to adjust your logic to search directly from the database, does that make sense?

Reply:
@jkappler

Reply:
@guilhermedcamacho, first, thanks for the review! 

Yes, you are right, my bad. I changed the code so that it checks the values from the DB. I had to use a dslQuery for it, because I couldn't make 

```
public long getObjectEntriesCount(
			long groupId, ObjectDefinition objectDefinition,
			Predicate predicate)
		throws PortalException 
```

work. It always threw an exception because the column could not be joined. If there's an easier way to do it, please let me know and I will change the code accordingly.

Thanks again!